server/controllers/donations.js

```diff
@@ -58,253 +55,52 @@ module.exports = (app) => {
-
-        const donation = {
-          UserId: user.id,
-          GroupId: group.id,
-          currency: currency,
-          amount: amountInt,
-          title: `Donation to ${group.name}`
-        };
-
-        models.Donation.create(donation)
-          .then(donation => {
-            if (isSubscription) {
-              const subscription = {
-                amount: amountFloat,
-                currency,
-                interval,
-                stripeSubscriptionId: charge.id,
-                data: charge
-              };
-              return donation
-                .createSubscription(subscription)
-                .then(() => cb(null, donation));
-            } else {
-              return cb(null, donation);
-            }
-          })
-          .catch(cb);
-      }],
-
-      retrieveBalanceTransaction: ['createCharge', (cb, results) => {
-        if (isSubscription) {
-          return cb();
-        } else {
-          const charge = results.createCharge;
-          gateways.stripe.retrieveBalanceTransaction(results.getGroupStripeAccount, charge.balance_transaction)
-          .then(balanceTransaction => cb(null, balanceTransaction))
-          .catch(cb);
-        }
-      }],
-
-      // Create the first transaction associated with that Donation, if this is not a subscription
-      createTransaction: ['createDonation', 'retrieveBalanceTransaction', (cb, results) => {
-
-        // If this is a subscription, wait for webhook to create a Transaction
-        if (isSubscription) {
-          return cb();
-        }
-
-        // Create a transaction for this one-time payment
-        const user = req.user;
-        const charge = results.createCharge;
-        const paymentMethod = results.getOrCreatePaymentMethod;
-        const balanceTransaction = results.retrieveBalanceTransaction
-        const fees = gateways.stripe.extractFees(balanceTransaction);
-        const hostFeePercent = group.hostFeePercent;
-        const payload = {
-          user,
-          group,
-          paymentMethod
-        };
-
-        payload.transaction = {
-          type: constants.type.DONATION,
-          DonationId: results.createDonation.id,
+    var paymentMethod;
+
+    // fetch Stripe Account and get or create Payment Method
+    return Promise.all([
+      req.group.getStripeAccount(),
+      models.PaymentMethod.getOrCreate({
+        token: payment.stripeToken,
+        service: 'stripe',
+        UserId: user.id })
+      ])
+    .then(results => {
+      const stripeAccount = results[0];
+      if (!stripeAccount || !stripeAccount.accessToken) {
+        return Promise.reject(new errors.BadRequest(`The host for the collective id ${req.group.id} has no Stripe account set up`));
+      } else if (process.env.NODE_ENV !== 'production' && _.contains(stripeAccount.accessToken, 'live')) {
+        return Promise.reject(new errors.BadRequest(`You can't use a Stripe live key on ${process.env.NODE_ENV}`));
```

Comment:
I was wondering about how to handle errors properly within promises. 
Is this the best way?
What about simply throwing the error: `throw new errors.BadRequest()`?